metals/src/main/scala/scala/meta/internal/metals/codeactions/BracelessBracefulSwitchCodeAction.scala

```diff
@@ -0,0 +1,241 @@
+  ) = {
+    if (hasBraces(braceableTree, document)) {
+      if (braceableTree.allowBracelessSyntax) {
+        // TODO `canUseBracelessSyntax` should be replaced by`canBeSyntacticallyBraceless`
+        // TODO Important: autoIndentDocument()
+        val braceableBranchLSPPos = braceableBranch.pos.toLSP
+        val braceableBranchStart = braceableBranchLSPPos.getStart
+        val braceableBranchEnd = braceableBranchLSPPos.getEnd
+        val startTextEdit = new TextEdit(
+          new l.Range(braceableBranchStart, braceableBranchStart),
+          bracelessStart
+        )
+        val endTextEdit = new TextEdit(
+          new l.Range(braceableBranchEnd, braceableBranchEnd),
+          bracelessEnd
+        )
+        val codeAction = new l.CodeAction()
+        codeAction.setTitle(BracelessBracefulSwitchCodeAction.goBraceless)
+        codeAction.setKind(this.kind)
+        codeAction.setEdit(
+          new l.WorkspaceEdit(
+            Map(
+              path.toURI.toString -> List(startTextEdit, endTextEdit).asJava
+            ).asJava
+          )
+        )
+        Some(codeAction)
+      } else {
+        None
+      }
+    } else {
+      // TODO Important: autoIndentDocument()
+      val braceableBranchLSPPos = braceableBranch.pos.toLSP
+      val braceableBranchStart = braceableBranchLSPPos.getStart
+      val braceableBranchEnd = braceableBranchLSPPos.getEnd
+      val startBraceTextEdit = new TextEdit(
+        new l.Range(braceableBranchStart, braceableBranchStart),
+        "{"
+      )
+      val endBraceTextEdit = new TextEdit(
+        new l.Range(braceableBranchEnd, braceableBranchEnd),
+        s"""|${document(braceableBranch.pos.end)}
+            |${getIndentationForPositionInDocument(braceableBranch.pos, document)}}""".stripMargin
+      )
+      val codeAction = new l.CodeAction()
+      codeAction.setTitle(BracelessBracefulSwitchCodeAction.goBraceFul)
+      codeAction.setKind(this.kind)
+      codeAction.setEdit(
+        new l.WorkspaceEdit(
+          Map(
+            path.toURI.toString -> List(
+              startBraceTextEdit,
+              endBraceTextEdit
+            ).asJava
+          ).asJava
+        )
+      )
+      Some(codeAction)
+    }
+  }
+
+  override def contribute(params: CodeActionParams, token: CancelToken)(implicit
+      ec: ExecutionContext
+  ): Future[Seq[l.CodeAction]] = Future {
+    val path = params.getTextDocument().getUri().toAbsolutePath
+    val range = params.getRange()
+    val maybeTree =
+      if (range.getStart == range.getEnd)
+        trees
+          .findLastEnclosingAt[Tree](
+            path,
+            range.getStart(),
+            applyWithSingleFunction
+          )
+      else None
+
+    val result =
+      for {
+        tree <- maybeTree
+        document <- buffers.get(path)
+      } yield {
+        tree match {
+          case _: Pkg => None
+          case classDefn: Defn.Class =>
+            createCodeActionForBraceableTree(
```

Comment:
We could probably reduce a number of cases here, but we can wait until it's out of draft.

Reply:
Yes, the problem is to get a hold of the value name to pass it to the  function…